src/puppetlabs/trapperkeeper/services/scheduler/scheduler_service.clj

```diff
@@ -30,27 +57,49 @@
     (log/debug "Initializing Scheduler Service")
     (let [pool (core/create-pool)]
       (assoc context :pool pool
-                     :jobs (atom #{}))))
+                     :jobs (atom #{})
+                     :after-id (atom 0))))
 
   (stop [this context]
     (log/debug "Shutting down Scheduler Service")
     ; Stop any jobs that are still running
     (core/stop-all-jobs! @(:jobs context) (get-pool this))
+    (reset! (:jobs context) #{})
```

Comment:
Since the `init` function does this ...
```clj
(assoc context :pool pool
                     :jobs (atom #{})
                     :after-id (atom 0))
```
I don't think this line is strictly necessary or a memory leak.  But no big deal, I don't mind leaving it, just wanted to understand.